content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -60,6 +61,7 @@ meta:
 - `meta` (optional): Arbitrary metadata can be added manually with this field.
   Any YAML contents is supported. `meta` contents are ignored by DVC, but they
   can be meaningful for user processes that read `.dvc` files.
+- `desc` (optional): Arbitrary string description for this `.dvc` file.
```

Comment:
Wait, so this field is for the whole .dvc file? So what's the difference with `meta`? I thought this would be per output (so, in the next bullet list).

Reply:
Yes, both. `meta` is not used by us at all, it is a completely arbitrary metadata. `desc` is required to be a string/text and will be used by Viewer and maybe even dvc at some point.

Reply:
I see, thanks for the explanation @efiop. Would it be better to keep it hidden (undocumented) for now? I see no significative difference between `meta` and `desc` from the DVC user perspective. Possible source of confusion

> p.s. I'd even suggest the Viewed uses a special format in `meta` values instead.

Reply:
Hey, I didn't know about this but it sounds like a good idea. Regarding screen real estate we can only really display it when you hover or click a file version, where we show some more long form information like the full hash, delta and change dates.

@shcheklein what do you think?

Reply:
We've talked about `meta` before. `meta` is for arbitrary data that is not for us at all. Not even for viewer. `desc` is something that we recognize and something that viewer will be using very soon. We will be using it in some dvc commands in the future as well.

Reply:
@fabiosantoscode yes, we should def read and show this information if it's present. I hoped it was already part of the tooltip in the viewer?

Reply:
> desc is something that we recognize

But https://dvc.org/doc/user-guide/dvc-files-and-directories reads "This doesn't affect any DVC operations." so it's only the viewer that recognizes it (and to my mind it could do the same with a special format in `meta` but up to you guys. I suspect at some point the Viewer may also display `meta` for the user, since it's their info after all — so the distinction becomes blurry again.

My main objection is to have both these options documented since they look like the same from the user perspective (and desc is a little inconsistent as mentioned in https://github.com/iterative/dvc.org/pull/1951#pullrequestreview-540675703 below).

Reply:
@jorgeorpinel It doesn't affect any dvc operations right now, but we still recognize it as a stage/output description string. Meta is also recognized by us but as an arbitrary object. Viewer will likely show it, but dvc - likely not. Still, those are separate things. We've considered using meta for it and decided not to.

Reply:
Not at all! I didn't know about this and it wasn't present in the designs

Reply:
> but we still recognize it as a stage/output description string

What does it mean to "recognize it"? Does DVC actually read/validate/process it in some way it doesn't do with `meta`? I do realize add/run have the --desc flag that writes it (except for stage outputs) so if that's what we mean then OK.

> We've considered using meta for it and decided not to.

Sure, I'm not insisting. I'm just detecting possible inconsistencies and confusions. Maybe I'll just extract this to an issue so we can officially burry it as a nice-to-have 😋

Thanks for all the answers!

Reply:
> What does it mean to "recognize it"? Does DVC actually read/validate/process it in some way it doesn't do with meta? I do realize add/run have the --desc flag that writes it (except for stage outputs) so if that's what we mean then OK.

Yes, it does read/validate it as str. Meta is arbitrary, we don't validate it ever.